Tools/TestWebKitAPI/Tests/JavaScriptCore/glib/TestJSC.cpp

```diff
@@ -3963,6 +4069,70 @@ static void testJSCPromises()
         g_assert_true(jsc_value_is_number(value.get()));
         g_assert_cmpint(jsc_value_to_int32(value.get()), ==, -1);
     }
+
+    {
+        GMainLoop* m = g_main_loop_new(NULL, FALSE);
+
+        GRefPtr<JSCContext> context = adoptGRef(jsc_context_new());
+
+        JSCClass* foooClass = jsc_context_register_class(context.get(), "Fooo", nullptr, nullptr, reinterpret_cast<GDestroyNotify>(foooFree));
+        GRefPtr<JSCValue> foooConstructor = adoptGRef(jsc_class_add_constructor(foooClass, nullptr, G_CALLBACK(foooCreate), m, nullptr, G_TYPE_POINTER, 0, G_TYPE_NONE));
+        jsc_class_add_method(foooClass, "getAsyncResolvedPromise", G_CALLBACK(getAsyncResolvedPromise), nullptr, nullptr, JSC_TYPE_VALUE, 0, G_TYPE_NONE);
+        jsc_class_add_method(foooClass, "getAsyncRejectedPromise", G_CALLBACK(getAsyncRejectedPromise), nullptr, nullptr, JSC_TYPE_VALUE, 0, G_TYPE_NONE);
+        jsc_class_add_method(foooClass, "getResolvedPromise", G_CALLBACK(getResolvedPromise), nullptr, nullptr, JSC_TYPE_VALUE, 0, G_TYPE_NONE);
+        jsc_class_add_method(foooClass, "getRejectedPromise", G_CALLBACK(getRejectedPromise), nullptr, nullptr, JSC_TYPE_VALUE, 0, G_TYPE_NONE);
+        jsc_class_add_method(foooClass, "getExceptionPromise", G_CALLBACK(getExceptionPromise), nullptr, nullptr, JSC_TYPE_VALUE, 0, G_TYPE_NONE);
```

Comment:
Why do we need a new class for this test? and why subtests are actually methods of the class?

Reply:
There is no particular reason, I wrote the tests based on what we did for our custom js api

Reply:
Ok I removed the class